test/integration/02-user-wallet/02-send-onchain.spec.ts

```diff
@@ -198,144 +300,192 @@ describe("UserWallet - onChainPay", () => {
+  )
+  expect(pendingTxsSender.length).toBe(0)
+
+  const settledTxsSender = txsSender.slice.filter(
+    ({ status, initiationVia, settlementVia, memo: txMemo }) =>
+      status === TxStatus.Success &&
+      initiationVia.type === PaymentInitiationMethod.OnChain &&
+      settlementVia.type === SettlementMethod.IntraLedger &&
+      txMemo === memo,
+  )
+  expect(settledTxsSender.length).toBe(1)
+  const senderSettledTx = settledTxsSender[0]
+
+  expect(senderSettledTx.settlementFee).toBe(0)
+  expect(senderSettledTx.settlementAmount).toBe(-senderAmount)
+  expect(senderSettledTx.displayCurrencyPerSettlementCurrencyUnit).toBeGreaterThan(0)
+
+  // Check txn details for received wallet
+  // ===
+  const { result: txsRecipient, error: errorUserA } =
+    await Wallets.getTransactionsForWalletId({
+      walletId: recipientWalletId,
+    })
+  if (errorUserA instanceof Error || txsRecipient === null) {
+    return errorUserA
+  }
+  const pendingTxsRecipient = txsRecipient.slice.filter(
+    ({ status }) => status === TxStatus.Pending,
+  )
+  expect(pendingTxsRecipient.length).toBe(0)
 
-      const feeRates = getFeesConfig()
-      const fee = feeRates.withdrawDefaultMin + 7050
+  const settledTxsRecipient = txsRecipient.slice.filter(
+    ({ status, initiationVia, settlementVia }) =>
+      status === TxStatus.Success &&
+      initiationVia.type === PaymentInitiationMethod.OnChain &&
+      settlementVia.type === SettlementMethod.IntraLedger,
+  )
+  const recipientSettledTx = settledTxsRecipient[0]
+
+  expect(recipientSettledTx.settlementFee).toBe(0)
+  expect(recipientSettledTx.settlementAmount).toBe(recipientAmount)
+  expect(recipientSettledTx.displayCurrencyPerSettlementCurrencyUnit).toBeGreaterThan(0)
+
+  // Check memos
+  // ===
+  const matchTx = (tx: WalletTransaction) =>
+    tx.initiationVia.type === PaymentInitiationMethod.OnChain &&
+    tx.initiationVia.address === address
+
+  // sender should know memo
+  const filteredTxs = txsSender.slice.filter(matchTx)
+  expect(filteredTxs.length).toBe(1)
+  expect(filteredTxs[0].memo).toBe(memo)
+
+  // receiver should not know memo from sender
+  const filteredTxsUserD = txsRecipient.slice.filter(matchTx)
+  expect(filteredTxsUserD.length).toBe(1)
+  expect(filteredTxsUserD[0].memo).not.toBe(memo)
+}
+
+describe("BtcWallet - onChainPay", () => {
+  it("sends a successful payment", async () => {
+    const res = await testExternalSend({
+      senderAccount: accountA,
+      senderWalletId: walletIdA,
+      amount,
+      sendAll: false,
+    })
+    if (res instanceof Error) throw res
+  })
 
-      const finalBalance = await getBalanceHelper(walletIdE)
-      expect(finalBalance).toBe(0)
+  it("sends all in a successful payment", async () => {
+    const res = await testExternalSend({
+      senderAccount: accountE,
+      senderWalletId: walletIdE,
+      amount,
+      sendAll: true,
+    })
+    if (res instanceof Error) throw res
```

Comment:
same problem here, this is not a good test

Reply:
https://github.com/GaloyMoney/galoy/pull/2072#discussion_r1058587750